Framework/MDAlgorithms/src/ConvertMDHistoToMatrixWorkspace.cpp

```diff
@@ -222,7 +222,7 @@ void ConvertMDHistoToMatrixWorkspace::make1DWorkspace() {
     VMD wsCoord = start + dir * line.x[i];
 
     VMD inTargetCoord = transform->applyVMD(wsCoord);
-    outputWorkspace->dataX(0)[i] = inTargetCoord[id];
+    outputWorkspace->mutableX(0)[i] = inTargetCoord[id];
```

Comment:
Hmmm... this is because its type def'd and the compiler doesn't recognise they are compatible types even if I try to use `getBareArray` - 

Instead what I would do is something as follows so we are not asking for the array in ever loop iteration:
```c++
auto &mutableXVals = outputWorkspace->mutableX(0)

  for (size_t i = 0; i < line.x.size(); ++i) {
    // Coordinates in the workspace being plotted
    VMD wsCoord = start + dir * line.x[i];

    VMD inTargetCoord = transform->applyVMD(wsCoord);
    mutableXVals[i] = inTargetCoord[id];
  }
```